invokeai/backend/model_manager/load/load_base.py

```diff
@@ -35,6 +40,16 @@ def __exit__(self, *args: Any, **kwargs: Any) -> None:
         """Context exit."""
         self._locker.unlock()
 
+    @contextmanager
+    def model_on_device(self) -> Generator[Tuple[Optional[Dict[str, torch.Tensor]], AnyModel], None, None]:
+        """Return a tuple consisting of the model's state dict (if it exists) and the locked model on execution device."""
```

Comment:
A few more important details that we should add to this docstring:
- That the state_dict will always be in RAM, not the target device.
- That the state_dict should be treated as read-only.
- The intended uses for the returned state_dict, since this API is a little weird. 
- Why the state_dict might be `None`.

Reply:
Will do. Look for a new (hopefully last) commit later this evening.

Reply:
Done.